src/ctapipe_io_nectarcam/__init__.py

```diff
@@ -1187,14 +1189,14 @@ def fill_r0r1_camera_container(self, zfits_event):
             # print("GAIN SELECTED")
             selected_gain = np.where(has_high_gain, 0, 1)
             waveform = np.full(
-                (n_pixels, n_samples), fill, dtype=dtype
```

Comment:
I am not sure this is strictly necessary. I introduced that because image extractors in `ctapipe` support again both low- and high-gain data, but I am not sure we have to enforce waveforms to always be of shape (n_gain, n_pixel, n_sample). @vmarandon, could you comment on that please ?

Reply:
Hi @tibaldo ,
Yes, it was a mistake on my side. The charge extractors in `ctapipe` can now support non-gain selected data, but it is not mandatory (contrary to what I understood in the first place).